src/view.rs

```diff
@@ -152,23 +198,52 @@ pub trait View {
 
     /// Use this method to layout the view's children.
     /// Usually you'll do this by calling `LayoutCx::layout_node`
-    fn layout(&mut self, cx: &mut LayoutCx) -> Node;
+    fn layout(&mut self, cx: &mut LayoutCx) -> Node {
+        cx.layout_node(self.id(), true, |cx| {
+            let mut nodes = Vec::new();
+            self.for_each_child_mut(&mut |child| {
+                nodes.push(cx.layout_view(child));
+                false
+            });
+            nodes
+        })
+    }
 
-    /// You must implement this if your view has children.
-    ///
     /// Responsible for computing the layout of the view's children.
-    fn compute_layout(&mut self, _cx: &mut LayoutCx) -> Option<Rect> {
-        None
+    fn compute_layout(&mut self, cx: &mut LayoutCx) -> Option<Rect> {
+        let mut layout_rect: Option<Rect> = None;
+        self.for_each_child_mut(&mut |child| {
+            let child_layout = cx.compute_view_layout(child);
+            if let Some(rect) = layout_rect {
+                layout_rect = Some(rect.union(child_layout));
+            } else {
+                layout_rect = Some(child_layout);
+            }
+            false
+        });
+        layout_rect
     }
 
     /// Implement this to handle events and to pass them down to children
     ///
     /// Return true to stop the event from propagating to other views
-    fn event(&mut self, cx: &mut EventCx, id_path: Option<&[Id]>, event: Event) -> bool;
+    fn event(&mut self, cx: &mut EventCx, id_path: Option<&[Id]>, event: Event) -> bool {
+        let mut handled = false;
+        self.for_each_child_mut(&mut |child| {
+            handled |= cx.view_event(child, id_path, event.clone());
```

Comment:
We'll probably break out if the event is handled.

Reply:
Returning true breaks the loop here.

Reply:
ah right